### ISSUE TITLE
images cut

### ISSUE THREAD

**Issue by Richelch** [meduzik/deckbuilder#3]
hi there

i have been away from running your amazing system

i have run into a massive issue where the cards are not rendering fully
thought it was my code but it might be something else

here is an example of me running your example deck and im getting this

![Capture JPG;jkl;jk](https://github.com/user-attachments/assets/c94d579d-55d0-4cd3-a467-f83e7f5629a9)

**Comment by meduzik** [comment]
Hey, I think something with Chrome driver changed and the window size is off now. Could you try with the latest commit to see if this fixes it?

**Comment by Richelch** [comment]
will try now, just saw the newest upload

**Comment by Richelch** [comment]
It has fixed it a little
the preview on chrome is now working perfectly

when running the Build Deck in TTS, the cards are still displaced

![image](https://github.com/user-attachments/assets/08684d44-dd2f-4d18-af39-18e1ef8242f3)

**Comment by meduzik** [comment]
> It has fixed it a little the preview on chrome is now working perfectly
> 
> when running the Build Deck in TTS, the cards are still displaced
> 
> ![image](https://private-user-images.githubusercontent.com/165936572/361246927-08684d44-dd2f-4d18-af39-18e1ef8242f3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ2MDUxOTUsIm5iZiI6MTcyNDYwNDg5NSwicGF0aCI6Ii8xNjU5MzY1NzIvMzYxMjQ2OTI3LTA4Njg0ZDQ0LWRkMmYtNGQxOC1hZjM5LTE4ZTFlZjgyNDJmMy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwODI1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDgyNVQxNjU0NTVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00ZDU5YWNiY2ExZjNhNDU4NGExNzY5ZDg5YmFhYzIzNzkwYjRkYTYwZjhlNzA1YmE2MWEyOTExYjIxMjhhMjljJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.8ZhEHqUUeeJXB7q4lG5sm3LdsFWS5hsgcUcHEL3bXCQ)

Could you share the full log of the app after you attempt to import the deck? And also your cached png/html files?
![image](https://github.com/user-attachments/assets/53091af4-a943-460b-94bf-09d58a985609)
![image](https://github.com/user-attachments/assets/e5eecf54-6958-4449-90a5-4b1d566645e9)

**Comment by Richelch** [comment]
Starting server on port 17352
Open http://localhost:17352/?preview&deck=example/deck.xml for an example deck
REQUESTED BUILDING "examples/deck.xml"
[1]+ Downloading 1jETidoy2203QMhEvAmFUvFKOXSiyGVZkELJXIrjDBJo/example_cards
[1]- Completed in 0.6399514675140381 s
Rendering D:\Users\Relch\Desktop\cards\Rivals 2.0\deck code\deckbuilder-master\examples\.cache\56f40932d6d1b78bf0db5d006bb28d6f1292848f\mydeck.1.face.html...
Starting new Chrome process

DevTools listening on ws://127.0.0.1:53416/devtools/browser/8d9d8af9-5207-4eae-8d50-2a80bf30e36b
Requested window size is 1616x1347, including extra 16x147
Actual window size is 1616x1347
Rendering complete in 2.4017913341522217s
Rendering D:\Users\Relch\Desktop\cards\Rivals 2.0\deck code\deckbuilder-master\examples\.cache\56f40932d6d1b78bf0db5d006bb28d6f1292848f\mydeck.1.back.html...
Requested window size is 945x1020, including extra 16x147
Actual window size is 945x1020
Rendering complete in 0.14372539520263672s
BUILDING "examples/deck.xml" SUCCESSFULLY COMPLETED!
127.0.0.1 - - [25/Aug/2024 17:50:05] "GET /?preview&deck=examples/deck.xml HTTP/1.1" 200 -
127.0.0.1 - - [25/Aug/2024 17:50:05] "GET /img?src=D:\\Users\\Relch\\Desktop\\cards\\Rivals%202.0\\deck%20code\\deckbuilder-master\\examples\\.cache\\56f40932d6d1b78bf0db5d006bb28d6f1292848f\\mydeck.184d00a88758.png HTTP/1.1" 200 -
127.0.0.1 - - [25/Aug/2024 17:50:05] "GET /img?src=D:\\Users\\Relch\\Desktop\\cards\\Rivals%202.0\\deck%20code\\deckbuilder-master\\examples\\.cache\\56f40932d6d1b78bf0db5d006bb28d6f1292848f\\mydeck.2b8492b0ca63.png HTTP/1.1" 200 -
no 'deck' param
127.0.0.1 - - [25/Aug/2024 17:50:06] "GET /favicon.ico HTTP/1.1" 500 -
[13652:17684:0825/175010.145:ERROR:usb_descriptors.cc(146)] Failed to read length for configuration 0.
[13652:17684:0825/175010.146:ERROR:usb_descriptors.cc(105)] Failed to read all configuration descriptors. Expected 1, got 0.
[13652:17684:0825/175010.146:ERROR:device_event_log_impl.cc(196)] [17:50:10.146] USB: usb_device_win.cc:95 Failed to read descriptors from \\?\usb#vid_1b1c&pid_1b13#0d01e03fae3e180654883f34f5001942#{a5dcbf10-6530-11d2-901f-00c04fb951ed}.
REQUESTED BUILDING "Decks/JinxDeck2.xml"
[2]+ Downloading 13M9YFPVI4AllvWoRerr2wr2zZ4ImXDzQybRilVK0xO0/Jinx
[2]- Completed in 1.1330413818359375 s
Rendering D:\Users\Relch\Desktop\cards\Rivals 2.0\deck code\deckbuilder-master\Decks\.cache\1148e331783752091cb9627d7220281038750a37\Jinxdeck.1.face.html...
Requested window size is 8016x3747, including extra 16x147
Actual window size is 8016x3747
Rendering complete in 2.9025161266326904s
Rendering D:\Users\Relch\Desktop\cards\Rivals 2.0\deck code\deckbuilder-master\Decks\.cache\1148e331783752091cb9627d7220281038750a37\Jinxdeck.1.back.html...
Requested window size is 928x1347, including extra 16x147
Actual window size is 928x1347
Rendering complete in 0.2814638614654541s
BUILDING "Decks/JinxDeck2.xml" SUCCESSFULLY COMPLETED!
127.0.0.1 - - [25/Aug/2024 17:50:15] "GET /?preview&deck=Decks/JinxDeck2.xml HTTP/1.1" 200 -
127.0.0.1 - - [25/Aug/2024 17:50:15] "GET /img?src=D:\\Users\\Relch\\Desktop\\cards\\Rivals%202.0\\deck%20code\\deckbuilder-master\\Decks\\.cache\\1148e331783752091cb9627d7220281038750a37\\Jinxdeck.9448a3ca8279.png HTTP/1.1" 200 -
127.0.0.1 - - [25/Aug/2024 17:50:15] "GET /img?src=D:\\Users\\Relch\\Desktop\\cards\\Rivals%202.0\\deck%20code\\deckbuilder-master\\Decks\\.cache\\1148e331783752091cb9627d7220281038750a37\\Jinxdeck.05a165457819.png HTTP/1.1" 200 -
Created TensorFlow Lite XNNPACK delegate for CPU.
no 'deck' param
127.0.0.1 - - [25/Aug/2024 17:50:17] "GET /favicon.ico HTTP/1.1" 500 -
REQUESTED BUILDING "Decks/JinxDeck2.xml"
[3]+ Downloading 13M9YFPVI4AllvWoRerr2wr2zZ4ImXDzQybRilVK0xO0/Jinx
[3]- Completed in 0.33075952529907227 s
Rendering D:\Users\Relch\Desktop\cards\Rivals 2.0\deck code\deckbuilder-master\Decks\.cache\1148e331783752091cb9627d7220281038750a37\Jinxdeck.1.face.html...
Requested window size is 8016x3695, including extra 16x95
Actual window size is 8016x3695
Rendering complete in 2.633991241455078s
Rendering D:\Users\Relch\Desktop\cards\Rivals 2.0\deck code\deckbuilder-master\Decks\.cache\1148e331783752091cb9627d7220281038750a37\Jinxdeck.1.back.html...
Requested window size is 928x1347, including extra 16x147
Actual window size is 928x1347
Rendering complete in 0.22656893730163574s
BUILDING "Decks/JinxDeck2.xml" SUCCESSFULLY COMPLETED!
127.0.0.1 - - [25/Aug/2024 17:50:27] "GET /?deck=Decks/JinxDeck2.xml HTTP/1.1" 200 -
[13652:17684:0825/193953.274:ERROR:device_event_log_impl.cc(196)] [19:39:53.274] USB: usb_service_win.cc:105 SetupDiGetDeviceProperty({{A45C254E-DF1C-4EFD-8020-67D146A850E0}, 6}) failed: Element not found. (0x490)


[Jinxdeck.1.face.txt](https://github.com/user-attachments/files/16741631/Jinxdeck.1.face.txt)

**Comment by meduzik** [comment]
Thanks! Could you try again with the new version? You may need to install Pillow library for Python if you do not have it yet (either `python -m pip install Pillow` or `python -m pip install -r requirements.txt`).

**Comment by Richelch** [comment]
works like a charm

what was the issue?

**Comment by meduzik** [comment]
I have to set Chrome window size to take a deck "screenshot" at the correct resolution, previously it treated the parameters as page size, but looks like it now treats them as outer window size, including title bars and such, so I had to adjust it by that amount. And if I did not "reset" it to an empty page between renderings it reported weird numbers for its non-page area causing different sheets to be cut differently. Fun case of software breaking after an update.
